pkg/controller/elasticsearch/driver/driver.go

```diff
@@ -523,3 +539,18 @@ func esReachableConditionMessage(internalService *corev1.Service, isServiceReady
 		return fmt.Sprintf("Service %s/%s has endpoints", internalService.Namespace, internalService.Name)
 	}
 }
+
+func isEsClientUpToDate(esClient esclient.Client, version version.Version, user esclient.BasicAuth, url string, caCerts []*x509.Certificate) bool {
```

Comment:
Feels like this is almost identical to the existing `Equals` function on the client and maybe should be defined in the client as well to avoid the two implementations from diverging and introducing subtle bugs i.e. this function returning the same client but the `Equals` function returning a different result.